tests/main/interfaces-appstream-metadata/task.yaml

```diff
@@ -0,0 +1,59 @@
+summary: The appstream-metadata interface grants access to package metadata
+
+details: |
+    A number of Linux distributions use the AppStream format to
+    provide metadata about both installed and available packages.
+
+    The appstream-metadata interface makes this information available
+    to a confined application by creating bind mounts from the host
+    system to equivalent points in the sandbox.  Together with an
+    interface granting access to the host system packaging system
+    (e.g. via PackageKit), it is possible to confine a graphical
+    package manager.
+
+systems: [-ubuntu-core-*]
+
+prepare: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
+
+restore: |
+    rm -f /usr/share/metainfo/test1.metainfo.xml
+    rm -f /usr/share/appdata/test2.metainfo.xml
+    rm -f /usr/share/app-info/xmls/test1.xml
+    rm -f /var/cache/app-info/xmls/test2.xml
+    rm -f /var/lib/app-info/yaml/test3.yml.gz
+    rm -f /var/lib/apt/lists/test3.yml.gz
+
+execute: |
+    mkdir -p /usr/share/metainfo
```

Comment:
This is something that would probably make more sense in prepare. All the way until "the plug is disconnected by default"